src/player.h

```diff
@@ -1564,6 +1564,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
          */
         void set_targeting_data( const targeting_data &td );
 
+        /**Pointer to player's home base*/
+        base_home* home;
```

Comment:
That's a bad way of doing this.
Instead, you want to specify ID of the owner in base's structure.

Reply:
There already is an owner ID in base...  the point of the pointer (just wanted to say that) was that `iexamine` functions are passed a reference to player  so referencing the base becomes really easy this way.  The player is going to at least want a boolean "has_base" to prevent creating a slowdown opening the construction menu since they should be limited to one home base. (future plans for smaller outposts)

Reply:
I.e. so I can do this: `auto &base = *p.home` in `iexamine.cpp`

Reply:
> The player is going to at least want a boolean "has_base" to prevent creating a slowdown opening the construction menu since they should be limited to one home base.

The problem is, you have no way of properly setting it. The code in `iexamine.cpp` will only be processed when examining, so not after save load.
In order to find player's base after loading the game, you'd have to scan all overmaps in existence to find it. And overmaps may actually disappear in some cases, so simply saving a `bool` would be buggy.

If you want to refer to exactly one base, you should save its origin. If the base doesn't exist, you want it set to `tripoint_min` and comment that `tripoint_min` means there is no base. That way you can later (for example, when saving or loading the game) check if that base really exists.

Reply:
> If the base doesn't exist, 

You should use a `cata::optional<tripoint>` instead of some arbitrary magic value that requires documentation writing, reading and remembering